common/strings.c

```diff
@@ -88,6 +88,91 @@ TDNFSafeAllocateString(
     goto cleanup;
 }
 
+size_t
+TDNFStringSepCount(
+    char *pszBuf,
+    char *pszSep
+    )
+{
+    size_t nCount = 0;
+    const char *pszTemp;
+
+    if (IsNullOrEmptyString(pszBuf))
+    {
+        BAIL_ON_TDNF_ERROR(1);
+    }
+    if (IsNullOrEmptyString(pszSep))
+    {
+        nCount = 1;
+        BAIL_ON_TDNF_ERROR(1);
```

Comment:
Here we should use a macros as defined in `include/tdnferror.h `.

Like:
```
dwError = ERROR_TDNF_INVALID_PARAMETER
BAIL_ON_TDNF_ERROR(dwError);
```

Reply:
Is this resolved? Current changes still shows we passing hardcoded value.

Reply:
Actually `BAIL_ON_TDNF_ERROR` here is misleading. It must just be `goto exit;` as this method does not return have an erroring scenario (see thread above).

```c
if (IsNullOrEmptyString(pszBuf))
{
    goto exit;
}
if (IsNullOrEmptyString(pszSep))
{
    nCount = 1;
    goto exit;
}
```
I am not sure if tDNF allows bare word goto to be placed instead of a `BAIL_ON_XXX()`. If it does, I'll go ahead with that.